pkg/controllers/providers/instancetype/capacity/suite_test.go

```diff
@@ -174,4 +209,65 @@ var _ = Describe("CapacityCache", func() {
 		mem.Sub(resource.MustParse(fmt.Sprintf("%dMi", int64(math.Ceil(float64(mem.Value())*options.FromContext(ctx).VMMemoryOverheadPercent/1024/1024)))))
 		Expect(i.Capacity.Memory().Value()).To(Equal(mem.Value()), "Expected capacity to match VMMemoryOverheadPercent calculation")
 	})
+
+	It("should properly update discovered capacity when matching AMI is not the first in the list", func() {
+		// Update nodeClass AMIs for this test
+		nodeClass.Status.AMIs = []v1.AMI{standardAMI}
+		ExpectApplied(ctx, env.Client, nodeClass)
+
+		// Get available instance types from the test environment
+		availableInstanceTypes, err := awsEnv.InstanceTypesProvider.List(ctx, nodeClass)
+		Expect(err).To(BeNil())
+		Expect(availableInstanceTypes).ToNot(BeEmpty(), "No instance types available in test environment")
+
+		// Choose the first instance type for testing
+		testInstanceType := availableInstanceTypes[0]
+		instanceTypeName := testInstanceType.Name
+
+		fmt.Printf("Initial memory capacity for %s: %v\n", testInstanceType.Name, testInstanceType.Capacity.Memory().Value())
```

Comment:
I assume this is left over from debugging, we should be good to drop this print now right?